decidim-core/lib/decidim/content_processor.rb

```diff
@@ -150,5 +150,30 @@ def self.parser_klass(type)
     def self.renderer_klass(type)
       "Decidim::ContentRenderers::#{type.to_s.camelize}Renderer"
     end
+
+    module Common
+      # Reports if the content being processed is in HTML format. The content
+      # is interpreted as HTML when it contains one or more HTML tags in it.
+      #
+      # @return [Boolean] a boolean indicating if the content is HTML
+      def html_content?(fragment = html_fragment)
+        return false if fragment.children.count.zero?
+        return true if fragment.children.count > 1
+
+        fragment.children.first.name != "text"
+      end
+
+      # Turns the content string into a document fragment object. This is useful
+      # for parsing the HTML content.
+      #
+      # @return [Loofah::HTML::DocumentFragment]
+      def html_fragment(text = nil)
+        if text
```

Comment:
Maybe we could simplify to some like (and avoid multiple parsings when we have text):  
```
@fragment ||= Loofah.fragment(text || content)
```

Reply:
No. That would always store the first content passed to the method meaning this would change the behavior and break things.

Only the content fragment is stored but any arbitrary text passed to this method should not be stored / would not make sense to store.